pkg/ibm_plugin/sdk/vpn.go

```diff
@@ -0,0 +1,596 @@
+// readyOrDeleted - if set to true polls VPN status until ready, else until deleted.
+func (c *CloudClient) pollVPNStatus(vpnId string, readyOrDeleted bool) error {
+	attempts := 40
+	sleepDuration := 10 * time.Second
+	utils.Log.Printf("\nPolling VPN status. Process might take up to %v seconds", attempts*(int(sleepDuration/time.Second)))
+	for attempt := 1; attempt <= attempts; attempt += 1 {
+
+		vpnData, _, err := c.vpcService.GetVPNGateway(c.vpcService.NewGetVPNGatewayOptions(
+			vpnId,
+		))
+
+		if err != nil {
+			if readyOrDeleted { // received err while waiting for ready status
+				return err
+			} else {
+				return nil // VPN can't be found, since it was deleted
+			}
+		}
+
+		// VPN desired status is "ready" and so is its current status
+		if readyOrDeleted && *vpnData.(*vpcv1.VPNGateway).Status == vpcv1.VPNGatewayStatusAvailableConst {
+			utils.Log.Printf("\nVPN achieved status ready in attempt No. %v\n", attempt)
+			return nil
+		}
+		time.Sleep(sleepDuration)
+	}
+	return fmt.Errorf("\nVPN with ID: %v hasn't achieved desired status in the alloted time frame", vpnId)
+}
+
+// returns the public IPs of a VPN
+// Note: route based VPN gateway uses the tunnel with the smaller public IP as the primary egress path if both tunnels are active.
+func (c *CloudClient) GetVPNIPs(vpnId string) ([]string, error) {
+	vpnData, _, err := c.vpcService.GetVPNGateway(c.vpcService.NewGetVPNGatewayOptions(
+		vpnId,
+	))
+	if err != nil {
+		return nil, err
+	}
+	vpnMembers := vpnData.(*vpcv1.VPNGateway).Members
+
+	publicIPs := make([]string, len(vpnMembers))
+	for i, member := range vpnMembers {
+		publicIPs[i] = *member.PublicIP.Address
+	}
+
+	return publicIPs, nil
+}
+
+// Creating routing table routes to VPN gateway connection.
+// routes from each of the VPC's zones will redirect traffic destined to any of the specified CIDRs to the VPN tunnel.
+// Idempotent function, i.e., err not raised if route already exists.
+func (c *CloudClient) createRoutes(routingTableID, vpcID, VPNConnectionID string, destinationCIDRs []string) error {
+	zones := ibmCommon.GetZonesOfRegion(c.region)
+
+	// create routes redirecting traffic to each destination CIDR
+	for _, destinationCIDR := range destinationCIDRs {
+		// TODO(cohen-j-omer) will extend to contain proper verification
+		if !strings.Contains(destinationCIDR, "/") { // convert ip provided to a CIDR
+			destinationCIDR += "/32"
+		}
+
+		// create a route from each zone to the specified connection
+		for _, zone := range zones {
+			routeConfig := &vpcv1.CreateVPCRoutingTableRouteOptions{
+				VPCID:          &vpcID,
+				RoutingTableID: &routingTableID,
+				Destination:    &destinationCIDR,
+				Zone: &vpcv1.ZoneIdentityByName{
+					Name: &zone,
+				},
+				Action: core.StringPtr(vpcv1.CreateVPCRoutingTableRouteOptionsActionDeliverConst),
+				Name:   core.StringPtr(GenerateResourceName(string(routeType))),
+				NextHop: &vpcv1.RoutePrototypeNextHop{
+					ID: &VPNConnectionID,
+				},
+			}
```

Comment:
Alignment seems off, make sure to lint properly

Reply:
`make lint` didn't raise an issue.